lib/Targets/AIEVecToCpp/TranslateAIEVecToCpp.cpp

```diff
@@ -2520,18 +2671,48 @@ LogicalResult CppEmitter::emitAttribute(Location loc, Attribute attr) {
     }
     if (auto vType = dense.getType().dyn_cast<VectorType>()) {
       if (auto iType = vType.getElementType().dyn_cast<IntegerType>()) {
+        unsigned width = iType.getWidth();
         if (llvm::all_of(dense, [](const APInt &val) { return val == 0; })) {
-          os << "null_";
-          if (failed(emitType(loc, vType)))
-            return failure();
+          if (AIEML) {
+            os << "broadcast_zero_s";
+            os << width;
+          } else {
+            os << "null_";
+            if (failed(emitType(loc, vType)))
+              return failure();
+          }
           os << "()";
           return success();
         }
-        os << '{';
-        interleaveComma(dense, os, [&](const APInt &val) {
-          printInt(val, shouldMapToUnsigned(iType.getSignedness()));
-        });
-        os << '}';
+
+        if (AIEML) {
+          bool hasSameValue = false;
+          std::string firstValue = "";
```

Comment:
I find the hasSameDenseValue API quite confusing.
Perhaps instead we can do
if (dens.isSplat()) {
  // get the first value
  // broadcast it 
} else {
  // produce list { , }
}